SOOUM/SOOUM/Managers/LoginManager.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  LoginManager.swift
+//  SOOUM
+//
+//  Created by JDeoks on 9/22/24.
+//
+
+import Foundation
+import Security
+
+class LoginManager {
+    
+    enum KeyChainKey: String {
+        case accessToken
+        case refreshToken
+        case deviceId
+    }
+    
+    static let shared = LoginManager()
+    
+    private init() { }
+    
+    /// 디바이스의 고유 UUID를 생성. 이미 존재할 경우 기존의 deviceId를 그대로 사용함.
+    /// - Returns: 키체인에 UUID 저장이 성공하면 deviceId를 반환하고, 없으면 `nil`을 반환함.
+    func initDeviceId() -> String? {
+        if let currentDeviceID = getTokenFromKeychain(tokenType: .deviceId) {
+            return currentDeviceID
+        }
+        let deviceId = UUID().uuidString
+        guard let deviceIdData = deviceId.data(using: .utf8) else {
+            return nil
+        }
+        
+        let query: [String: Any] = [
+            kSecClass as String: kSecClassGenericPassword,
+            kSecAttrAccount as String: KeyChainKey.deviceId.rawValue,
+            kSecValueData as String: deviceIdData
+        ]
```

Comment:
```suggestion
        let query: [String: Any] = [
            kSecClass as String: kSecClassGenericPassword,
            kSecAttrAccount as String: KeyChainKey.deviceId.rawValue,
            kSecAttrService as String: self.service,
            kSecValueData as String: deviceIdData
        ]
```

keychain은 디바이스 내에서 공유되기 때문에 식별할 수 있는 id가 필요합니다.
`private let service = "com.sooum"` 추천합니다.